### PR TITLE
PD-3927   Multi-frequency and time mode support for Python package

### PR DESCRIPTION
**Pull request by pstarkov**

### CODE REVIEW

File: tests/test_knoema_data_time.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    def setUp(self):
+        apicfg = knoema.ApiConfig()
+        apicfg.host = os.environ['BASE_HOST'] if 'BASE_HOST' in os.environ else self.base_host
+        apicfg.app_id = os.environ['KNOEMA_APP_ID'] if 'KNOEMA_APP_ID' in os.environ else ''
+        apicfg.app_secret = os.environ['KNOEMA_APP_SECRET'] if 'KNOEMA_APP_SECRET' in os.environ else ''
+
+    def test_getdata_by_TransformationDataReader_timemembers(self):
+        """The method is testing getting single series by dimension member ids"""
+
+        data_frame = knoema.get('IMFWEO2021Apr', Country='614', Subject='BCA', timemembers='1980;2002;2023')
```
Comment (SlavkaKnoema) [inline tests/test_knoema_data_time.py:21]: Why timemembers is string? Could we use array and internally join items to string?

Comment (pstarkov) [reply tests/test_knoema_data_time.py:21]: the same as for all parameters: several dimension members, multiple frequencies

File: knoema/__init__.py [hunk 2]
```diff
@@ -60,24 +60,68 @@ def get(dataset = None, include_metadata = False, mnemonics = None, transform =
...
+                dim_values['timemembers'] = time
+                advanced = True
+                ++time_params_count
             continue
 
         if len(value) > 0 and value[0].startswith('@'):
             has_agg = True
 
+    if not dataset:
+        raise ValueError('Dataset id is not specified')
```
Comment (aadilov) [inline knoema/__init__.py:117]: if you search by mnemonics without dataset specified you get this error
`data_frame = knoema.get(mnemonics = 'mnemonic1;mnemonic2')`

Comment (PavelPimenov) [reply knoema/__init__.py:117]: +

File: knoema/api_definitions.py [hunk 3]
```diff
@@ -112,9 +112,9 @@ def __init__(self, data):
             self.name_map[item.name.upper()] = item
             if 'id' in item.fields:
                 self.id_map[item.fields['id'].upper()] = item
-            if 'regionid' in item.fields:
+            if 'regionid' in item.fields and item.fields['regionid'] is not None:
```
Comment (PavelPimenov) [inline knoema/api_definitions.py:115]: How these changes belong to your ticket?

**Pull request merged**